course-instructions.md

```diff
@@ -0,0 +1,342 @@
+  ),
+  # Create a plot
+  tar_target(
+    figure_1_png,
+    make_plot(out_filepath = "figure_1.png", data = eval_data), 
+    format = "file"
+  ),
+  # Save the processed data
+  tar_target(
+    model_summary_results_csv,
+    write_csv(eval_data, file = "model_summary_results.csv"), 
+    format = "file"
+  ),
+  # Save the model diagnostics
+  tar_target(
+    model_diagnostic_text_txt,
+    generate_model_diagnostics(out_filepath = "model_diagnostic_text.txt", data = eval_data), 
+    format = "file"
+  )
+)
+```    
+
+then use `Ctrl+D` to exit the file creation mode and return to the prompt. 
+
+Finally, create a `pull request` that includes this new file (the file should be called `_targets.R`). In the following section, we'll provide some helpful suggestions to improve your pipeline.
+
+</details>
+
+<hr>
+
+<details>
+<summary><h2>Modifying Your Pipeline by Adding to Your Pull Request</h2></summary>
+
+* Great work, but it would be great to download the `model_RMSEs.csv` to `1_fetch/out/model_RMSEs.csv` instead. Can you make a change to make sure that happens when you build this target?
+* Like with the downloaded data, let's move this plot output file into the appropriate phase folder (so, something like `3_visualize/out/figure_1.png`). You can also rename the figure if you'd like to use something more descriptive. 
+
+<hr> 
+
+So, what does this do for you? Well, if you had this `_targets.R` file in your current working directory (which you do), and you had defined the `download_data()`, `process_data()`, `plot_data()`, and `generate_diagnostics()` functions in `code.R` (which you haven't), it would look like this when you ran `tar_make()` from the `targets` package:
+
+![targets run image](https://user-images.githubusercontent.com/13220910/128781588-64911332-6464-4cb3-b77d-eaa7063d6bc1.png)
+
+---
+
+Now - adding on to this PR by pushing up additional commits - add and modify code, folders, and this makefile so you can build _your_ `3_visualize/out/figure_1.png` with `targets::tar_make()` (BTW, if you haven't run into this syntax, {package_name}::{function_name} allows you to run a function without `library({package_name})`). Build on your existing folders, functions, and code from the previous sections by continuing with the phases `1_fetch`, `2_process`, and `3_visualize`. Comment in the pull request with a screenshot of the build message, like we have in the message above :point_up:. Assign your course contact to review your PR, and they may ask for a few changes before merging. 
+
+There is a file in your repo called `.gitignore`. Add a line with `_targets` so that git doesn't track the contents of the `targets` directory. In addition, make sure that the contents of `1_fetch/out/*` will be ignored by git because you don't want to store changes to your downloaded .csv files (you may have already added a more generic `*/out/*` to your `.gitignore` that handles all `out/` folders in this repo). 
```

Comment:
Wonder if this should be like "Whoops! Did you commit your local `_targets` folder? This folder should be ignored..."

Reply:
Often I am seeing people committing this first, and then not being able to uncommit it when they realize it should be ignored

Reply:
Yes, It think most people make a mistake here. `gitignore` first gets mentioned at [line 125](https://github.com/padilla410/ds-pipelines-targets-1-course-static/blob/main/course-instructions.md?plain=1#L125). I think it ultimately happens because [line 298](https://github.com/padilla410/ds-pipelines-targets-1-course-static/blob/b4133804c68e11abcf2cbd02482fff0cc6452f96/course-instructions.md?plain=1#L298) says "add and modify code, folders, and this makefile so you can build _your_ `3_visualize/out/figure_1.png` with `targets::tar_make()`".

I think the time to address the `_targets` subfolder is right after the first `tar_make` command.

Reply:
modified